src/parameterizations/lateral/MOM_self_attr_load.F90

```diff
@@ -0,0 +1,278 @@
+    HDat(2) = ( 2.0 / 3.0) * (H1 - L1)
+    LDat(2) = (-1.0 / 3.0) * (H1 - L1)
+    KDat(2) = (-1.0 / 3.0) * H1 - (2.0 / 3.0) * L1 - 1.0
+  endif
+
+  do m=0,nlm ; do n=m,nlm
+    l = order2index(m,nlm)
+    Love_Scaling(l+n-m) = (3.0 / real(2*n+1)) * (rhoW / rhoE) * (1.0 + KDat(n+1) - HDat(n+1))
+  enddo ; enddo
+end subroutine calc_love_scaling
+
+!> This subroutine initializes the self-attraction and loading control structure.
+subroutine SAL_init(G, US, param_file, CS)
+  type(ocean_grid_type),  intent(inout) :: G    !< The ocean's grid structure.
+  type(unit_scale_type),  intent(in)    :: US   !< A dimensional unit scaling type
+  type(param_file_type),  intent(in)    :: param_file !< A structure to parse for run-time parameters.
+  type(SAL_CS), intent(inout) :: CS   !< Self-attraction and loading control structure
+
+  ! Local variables
+# include "version_variable.h"
+  character(len=40)  :: mdl = "MOM_self_attr_load" ! This module's name.
+  integer :: lmax ! Total modes of the real spherical harmonics [nondim]
+  real :: rhoW    ! The average density of sea water [R ~> kg m-3].
+  real :: rhoE    ! The average density of Earth [R ~> kg m-3].
+
+  logical :: calculate_sal
+  logical :: tides, use_tidal_sal_file
+  real :: tide_sal_scalar_value
+
+  ! Read all relevant parameters and write them to the model log.
+  call log_version(param_file, mdl, version, "")
+
+  call get_param(param_file, '', "TIDES", tides, default=.false., do_not_log=.True.)
+  call get_param(param_file, mdl, "CALCULATE_SAL", calculate_sal, "If true, calculate "//&
+                 " self-attraction and loading.", default=tides, do_not_log=.True.)
+  if (.not. calculate_sal) return
+
+  if (tides) then
+    call get_param(param_file, '', "USE_PREVIOUS_TIDES", CS%use_tidal_sal_prev, &
+                   default=.false., do_not_log=.True.)
+    call get_param(param_file, '', "TIDAL_SAL_FROM_FILE", use_tidal_sal_file, &
+                   default=.false., do_not_log=.True.)
+  endif
+
+  call get_param(param_file, mdl, "SAL_SCALAR_APPROX", CS%use_sal_scalar, &
+                 "If true, use the scalar approximation to calculate self-attraction and"//&
+                 " loading.", default=tides .and. (.not.use_tidal_sal_file))
+  call get_param(param_file, mdl, "SAL_HARMONICS", CS%use_sal_sht, &
+                 "If true, use the online spherical harmonics method to calculate"//&
+                 " self-attraction and loading.", default=.false.)
+
+  if (CS%use_sal_scalar .or. CS%use_tidal_sal_prev) then
+    call get_param(param_file, '', "TIDE_SAL_SCALAR_VALUE", tide_sal_scalar_value, &
+                   units="m m-1", default=0.0, do_not_log=.True.)
+    if (tide_sal_scalar_value/=0.0) &
+      call MOM_error(WARNING, "TIDE_SAL_SCALAR_VALUE is a deprecated parameter. "//&
+                              "Use SAL_SCALAR_VALUE instead." )
+      call get_param(param_file, mdl, "SAL_SCALAR_VALUE", CS%sal_scalar_value, &
```

Comment:
Please correct the indenting on this line.  As written, the indenting makes it look like this is inside of the `if (tide_sal_scalar_value /= 0.0)` block.